java/service_instance.go

```diff
@@ -1639,11 +1664,73 @@ func (c *ServiceInstanceClient) ScaleUpDownServiceInstance(input *ScaleUpDownSer
 
 	// Wait for the service instance to be running and return the result
 	// Don't have to unqualify any objects, as the GetServiceInstance method will handle that.
-	serviceInstance, err := c.WaitForServiceInstanceRunning(getInput, c.PollInterval, c.Timeout)
+	serviceInstance, err := c.WaitForServiceInstanceState(getInput, ServiceInstanceLifecycleStateStart, c.PollInterval, c.Timeout)
 	// The service instance is returned as nil if it enters a terminating state.
 	if err != nil || serviceInstance == nil {
 		return fmt.Errorf("error creating service instance %q: %+v", input.Name, err)
 	}
 
 	return nil
 }
+
+// DesiredStateInput defines the attributes for how to set the desired state of a java service instance.
+type DesiredStateInput struct {
+	// Flag that specifies whether to control the entire service instance.
+	// This attribute is not applicable to the restart command.
+	// Optional
+	AllServiceHosts bool `json:"allServiceHosts,omitemtpy"`
+	// Groups properties for the Oracle WebLogic Server component (WLS).
+	// Optional
+	Components *DesiredStateComponent `json:"components,omitempty"`
+	// Name of the Java Cloud Service instance.
+	// Required.
+	Name string `json:"-"`
+	// Type of the request.
+	// Required
+	LifecycleState ServiceInstanceLifecycleState `json:"-"`
+}
+
+// DesiredStateComponent groups properties for the Oracle WebLogic Server component (WLS) or the Oracle
+// Traffice Director (OTD) component.
+type DesiredStateComponent struct {
+	// Properties for the Oracle Traffic Director (OTD) component.
+	// Optional
+	OTD *DesiredStateHost `json:"OTD,omitempty"`
+	// Properties for the Oracle WebLogic Server (WLS) component.
+	// Optional
+	WLS *DesiredStateHost `json:"WLS,omitempty"`
+}
+
+// DesiredStateHost defines the properties of the hosts
+type DesiredStateHost struct {
+	// A single host name. Only application cluster hosts can be specified.
```

Comment:
It says a single host name but takes an array?

Reply:
I was/am confused by it as well. This is from the API docs though so maybe the expect to add support for multiple hosts at a later date